picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -673,31 +673,21 @@ void shutdown() {
    * {@link Picasso} instance. You can either use this directly or by setting it as the global
    * instance with {@link #setSingletonInstance}.
    */
-  public static Picasso with(@NonNull Context context) {
-    if (context == null) {
-      throw new IllegalArgumentException("context == null");
-    }
+  public static Picasso with() {
     if (singleton == null) {
       synchronized (Picasso.class) {
         if (singleton == null) {
-          singleton = new Builder(context).build();
+          if (PicassoProvider.context == null) {
+            throw new IllegalStateException("context == null");
```

Comment:
Drive-by thought:
This would only happen if you remove the ContentProvider in your manifest, right?
Should the error message suggest checking for that?